clients/src/main/java/org/apache/kafka/common/utils/Time.java

```diff
@@ -86,4 +89,30 @@ default Timer timer(Duration timeout) {
         return timer(timeout.toMillis());
     }
 
+    /**
+     * Wait for a future to complete, or time out.
+     *
+     * @param future        The future to wait for.
+     * @param deadlineNs    The time in the future, in monotonic nanoseconds, to time out.
+     * @return              The result of the future.
+     * @param <T>           The type of the future.
+     */
+    default <T> T waitForFuture(
+        CompletableFuture<T> future,
+        long deadlineNs
```

Comment:
Does it make sense to use `Deadline` here?

Reply:
Can't use that here since `:clients` doesn't depend on `:server-common`.